### PR TITLE
(feat) O3-3176: Improvements to the overdue vitals tag

### PR DESCRIPTION
**Pull request by miirochristopher**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

Indicate how old the Vitals are in the red chip, in a narrative based format, for any vitals not were not collected during the current Active Visit.

Logic and format
From 1 days and up to 7 days - ‘These vitals are X days old’
Up to two weeks - ‘These vitals are over one week old’
Over two weeks - ‘These vitals are out of date.'

## Screenshots

Screenshot

Screenshot 1

Screenshot 3

Screenshot 2

Screenshot

## Related Issue
https://openmrs.atlassian.net/browse/O3-3176)

### CODE REVIEW

File: packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx [hunk 1]
```diff
@@ -58,6 +60,18 @@ const VitalsHeader: React.FC<VitalsHeaderProps> = ({ patientUuid }) => {
     const vitalsTakenToday = Boolean(dayjs(latestVitals?.date).isToday());
     const vitalsOverdue = hasActiveVisit && !vitalsTakenToday;
 
+    const now = dayjs();
+    const vitalsOverduePeriod = Math.round(dayjs.duration(now.diff(latestVitals?.date)).asDays());
+    let overdueVitalsTag: string = '';
+
+    if (vitalsOverduePeriod >= 1 && vitalsOverduePeriod <= 7) {
+      overdueVitalsTag = `These vitals are ${vitalsOverduePeriod} days old.`;
```
Comment (ojwanganto) [inline packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx:68]: Should we find something for the 1-day old text? The rest look good

Comment (miirochristopher) [reply packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx:70]: @ojwanganto 1-day old text added in the latest push. Thanks.

Comment (ojwanganto) [reply packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx:70]: Looks good. Thanks

**Review by ojwanganto — APPROVED** [review]

File: packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx [hunk 2]
```diff
@@ -58,6 +60,20 @@ const VitalsHeader: React.FC<VitalsHeaderProps> = ({ patientUuid }) => {
...
+
+    if (vitalsOverduePeriod == 1) {
+      overdueVitalsTag = 'These vitals are 1 day old.';
+    } else if (vitalsOverduePeriod >= 2 && vitalsOverduePeriod <= 7) {
+      overdueVitalsTag = `These vitals are ${vitalsOverduePeriod} days old.`;
+    } else if (vitalsOverduePeriod >= 8 && vitalsOverduePeriod <= 14) {
+      overdueVitalsTag = 'These vitals are over one week old.';
+    } else {
+      overdueVitalsTag = 'These vitals are out of date.';
+    }
```
Comment (denniskigen) [inline packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx:75]: This may work better:

```suggestion
  switch (true) {
      case vitalsOverduePeriod === 1:
        overdueVitalsTag = t('dayOldVitals', 'These vitals are one day old');
        break;
      case vitalsOverduePeriod >= 2 && vitalsOverduePeriod <= 7:
        overdueVitalsTag = t('severalDaysOldVitals', 'These vitals are {{vitalsOverduePeriod}} days old', {
          vitalsOverduePeriod,
        });
        break;
      case vitalsOverduePeriod >= 8 && vitalsOverduePeriod <= 14:
        overdueVitalsTag = t('overOneWeekOldVitals', 'These vitals are over one week old');
        break;
      default:
        overdueVitalsTag = t('outOfDateVitals', 'These vitals are out of date');
        break;
    }
```

Most importantly, we need these strings to be translated.

File: packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx [hunk 3]
```diff
@@ -71,7 +87,7 @@ const VitalsHeader: React.FC<VitalsHeaderProps> = ({ patientUuid }) => {
                 <Tag type="red">
                   <span className={styles.overdueIndicator}>
                     <Time />
-                    {t('overdue', 'Overdue')}
+                    {t('overdue', 'Overdue').concat(' : ', overdueVitalsTag)}
```
Comment (denniskigen) [inline packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx:90]:
```suggestion
                   {`${t('overdue', 'Overdue')}: ${overdueVitalsTag}`}
```

You might also want to add this property to the `overdueIndicator` class:

```scss
.overdueIndicator {
  // other styles...
  min-inline-size: max-content;
}

The `max-content` value ensures that the element will be as wide as its content, without wrapping.

**Review by denniskigen — COMMENTED** [review]
I've suggested some tweaks to the approach. Please try them out.

File: packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx (hunk 2 above)
Comment (denniskigen) [inline packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx:75]:
```suggestion
    switch (true) {
      case vitalsOverduePeriod >= 1 && vitalsOverduePeriod <= 7:
        overdueVitalsTag = t('daysOldVitals', 'These vitals are {{vitalsOverduePeriod}} days old', {
          vitalsOverduePeriod,
        });
        break;
      case vitalsOverduePeriod >= 8 && vitalsOverduePeriod <= 14:
        overdueVitalsTag = t('overOneWeekOldVitals', 'These vitals are over one week old');
        break;
      default:
        overdueVitalsTag = t('outOfDateVitals', 'These vitals are out of date');
        break;
    }
```

This might work better according to the criteria laid out in the ticket:

- From 1 day and up to 7 days - ‘These vitals are X days old’
- Up to two weeks - ‘These vitals are over one week old’
- Over two weeks - ‘These vitals are out of date.'

Also, we need to translate those strings.

**Comment by miirochristopher** [comment]
@denniskigen made the tweaks requested. Thanks

File: packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx [hunk 4]
```diff
@@ -57,6 +59,23 @@ const VitalsHeader: React.FC<VitalsHeaderProps> = ({ patientUuid }) => {
     const hasActiveVisit = Boolean(currentVisit?.uuid);
     const vitalsTakenToday = Boolean(dayjs(latestVitals?.date).isToday());
```
Comment (denniskigen) [inline packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.component.tsx:60]: We don't need to handle the case where vitals are collected today because it will be taken care of in this condition. The tag won't get rendered at all.

File: packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.test.tsx [hunk 5]
```diff
@@ -88,8 +88,7 @@ describe('VitalsHeader', () => {
     expect(getByTextWithMarkup(/Height\s*-\s*/i)).toBeInTheDocument();
     expect(getByTextWithMarkup(/BMI\s*-\s*/i)).toBeInTheDocument();
     expect(getByTextWithMarkup(/Weight\s*-\s*/i)).toBeInTheDocument();
-
-    expect(screen.getByText(/overdue/i)).toBeInTheDocument();
+    expect(screen.getByText(/overdue: these vitals are out of date/i)).toBeInTheDocument();
```
Comment (denniskigen) [inline packages/esm-patient-vitals-app/src/vitals-and-biometrics-header/vitals-header.test.tsx:91]: This just updates the expectation with the expected tag content.

File: packages/esm-patient-vitals-app/translations/en.json [hunk 6]
```diff
@@ -13,6 +13,8 @@
   "checkForValidity": "Some of the values entered are invalid",
   "date": "Date",
   "dateAndTime": "Date and time",
+  "daysOldVitals_one": "These vitals are {{count}} day old",
+  "daysOldVitals_other": "These vitals are {{count}} days old",
```
Comment (denniskigen) [inline packages/esm-patient-vitals-app/translations/en.json:17]: @miirochristopher we don't update these by hand. Rather, when you add new translation strings in your code, i18next-parser will automatically extract them from your code when you commit your changes. Updates to non-en locale files are handled automatically via the Transifex integration. 

tl;dr, don't update translation files manually.

**Comment by denniskigen** [comment]
I've pushed a commit updating things further and some comments that explain why I made the changes.

**Review by denniskigen — APPROVED** [review]
